From 0cfec21c4a02aef3c752b6aacd69511579d65a2a Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@apache.org>
Date: Fri, 10 Apr 2015 03:51:02 +0000
Subject: [PATCH 0836/1164] CDH-47499 : HIVE-10267 : HIVE-9664 makes hive
 depend on ivysettings.xml : trivial breakage fix
 (Sushanth Sowmyan, reviewed by Thejas Nair)

Change-Id: Icba8a9e5b9c1bf33c35552c728007395e0604587
git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1672535 13f79535-47bb-0310-9956-ffa450edef68
---
 .../hadoop/hive/ql/session/DependencyResolver.java |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/session/DependencyResolver.java b/ql/src/java/org/apache/hadoop/hive/ql/session/DependencyResolver.java
index 27bf3e4..b20e975 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/session/DependencyResolver.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/session/DependencyResolver.java
@@ -20,6 +20,7 @@
 
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.net.URL;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.LinkedList;
@@ -58,8 +59,11 @@ public DependencyResolver() {
 
     // If HIVE_HOME is not defined or file is not found in HIVE_HOME/conf then load default ivysettings.xml from class loader
     if (ivysettingsPath == null || !(new File(ivysettingsPath).exists())) {
-      ivysettingsPath = ClassLoader.getSystemResource("ivysettings.xml").getFile();
-      _console.printInfo("ivysettings.xml file not found in HIVE_HOME or HIVE_CONF_DIR," + ivysettingsPath + " will be used");
+      URL ivysetttingsResource = ClassLoader.getSystemResource("ivysettings.xml");
+      if (ivysetttingsResource != null){
+        ivysettingsPath = ivysetttingsResource.getFile();
+        _console.printInfo("ivysettings.xml file not found in HIVE_HOME or HIVE_CONF_DIR," + ivysettingsPath + " will be used");
+      }
     }
 
   }
-- 
1.7.9.5

